framework/core/image.cpp

```diff
@@ -78,18 +78,20 @@ ImagePtr ImageBuilder::build_unique(const Device &device) const
 	return std::make_unique<Image>(device, *this);
 }
 
-Image::Image(Device const         &device,
-             const VkExtent3D     &extent,
-             VkFormat              format,
-             VkImageUsageFlags     image_usage,
-             VmaMemoryUsage        memory_usage,
-             VkSampleCountFlagBits sample_count,
-             const uint32_t        mip_levels,
-             const uint32_t        array_layers,
-             VkImageTiling         tiling,
-             VkImageCreateFlags    flags,
-             uint32_t              num_queue_families,
-             const uint32_t       *queue_families) :
+Image::Image(Device const                       &device,
+             const VkExtent3D                   &extent,
+             VkFormat                            format,
+             VkImageUsageFlags                   image_usage,
+             VmaMemoryUsage                      memory_usage,
+             VkSampleCountFlagBits               sample_count,
+             const uint32_t                      mip_levels,
+             const uint32_t                      array_layers,
+             VkImageTiling                       tiling,
+             VkImageCreateFlags                  flags,
+             uint32_t                            num_queue_families,
+             const uint32_t                     *queue_families,
+             VkImageCompressionFlagsEXT          compression_flags,
+             VkImageCompressionFixedRateFlagsEXT fixed_rate_flags) :
```

Comment:
I'd argue against making this ctor any more complicated, and instead using the `ImageBuilder` logic directly in the example that's using it.

Reply:
Addressed in my last commit